scala/src/main/scala/com/xebia/functional/xef/scala/prompt/PromptTemplate.scala

```diff
@@ -0,0 +1,22 @@
+package com.xebia.functional.xef.scala.prompt
+
+import com.xebia.functional.xef.scala.auto.*
+import io.circe.Decoder
+
+trait PromptTemplate[A] {
+
+  def chain(template: String): A
```

Comment:
This functionality should be developed in Kotlin multiplatform and mapped to other lang bindings. @juanpedromoreno @necosta  What criteria are we following to decide what gets developed in Kotlin vs. Scala?

Reply:
Hello @raulraja, I agree with your observation. Nelson and I have had an offline discussion regarding this matter. The Scala API was unable to create prompts beyond utilizing string interpolations, unlike Kotlin does [here](https://github.com/xebia-functional/xef/blob/bd7e3bc8e8088682889e000d0c69ba9b025f2d1d/core/src/commonMain/kotlin/com/xebia/functional/xef/prompt/Prompt.kt). Due to this limitation, we ended up creating different designs without considering the possible divergence. I suggest we work on unifying the APIs once again 👍

Reply:
Will try to move this logic to Kotlin. Scala should follow the Kotlin API yes, it was just overlooked.